### PR TITLE
(DOCSP-27014 & DOCSP-27015): @realm/react-ify Handle Sync Errors page

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

@realm/react-ify Handle Sync Errors page. Consolidate Handle Sync Errors and Client Reset pages. 

### Jira

- https://jira.mongodb.org/browse/DOCSP-27014
- https://jira.mongodb.org/browse/DOCSP-27015

### Staged Changes

- [Handle Sync Errors (RN)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27014/sdk/react-native/sync-data/handle-sync-errors/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/sync-data/handle-sync-errors.txt [hunk 1]
```diff
@@ -10,20 +10,299 @@ Handle Sync Errors - React Native SDK
...
+- ``session``: :js-sdk:`Session <Realm.App.Sync.Session.html>` object representing
+  the state of the Device Sync session.
+- ``path``: String with the path to the current realm database file.
+
+The following example demonstrates how you can manually handle this error case by
+discarding all unsynced changes:
+
+.. literalinclude:: /examples/generated/node/client-reset.snippet.recovery-fallback.js
+   :language: js
+   :end-before: realm = await Realm.open(config);
```
Comment (mongodben) [inline source/sdk/react-native/sync-data/handle-sync-errors.txt:214]: @takameyer in the example, it talks about referencing the `realm` variable at the top of the example, but i'm not sure how this'd work in a react native context?

you have any ideas here?

**Review by krollins-mdb — COMMENTED** [review]
Combining these two pages makes a lot of sense. I think we could do a bit more to "realm reactify". For example, the samples are all functionally sound, but we don't provide an example of how the sync config object plugs into `<RealmProvider>`. Or mention `@realm/react`.

My suggestions are mostly minor and should help keep `@realm/react` in readers' minds.

File: source/sdk/react-native/sync-data/handle-sync-errors.txt [hunk 2]
```diff
@@ -10,20 +10,299 @@ Handle Sync Errors - React Native SDK
...
 While developing an application that uses Device Sync, you should set an error
 handler. This error handler will detect and respond to any failed sync-related
-API calls. 
+API calls.
 
-Set an error handler by registering an error callback as part of the :js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`.
+Set an error handler by registering an error callback as part of the
+:js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`.
 
 .. literalinclude:: /examples/generated/node/sync-changes-between-devices.snippet.sync-changes-between-devices-handle-sync-errors.js
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt:23]: yeah this seems like a mistake to me. will fix here.

File: source/sdk/react-native/sync-data/handle-sync-errors.txt [hunk 3]
```diff
@@ -10,20 +10,299 @@ Handle Sync Errors - React Native SDK
...
+If your app requires specific client reset logic that can't be handled automatically,
+you may want or need to :ref:`add a manual client reset handler <react-native-manually-recover-unsynced-changes>`
+to the automatic client reset mode.
+
+.. _react-native-client-reset-recovery:
+
+Client Reset with Recovery
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 10.23.0
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt]: good point. fixing for this page in this PR. for other pages, created separate ticket [DOCSP-28173](https://jira.mongodb.org/browse/DOCSP-28173)

File: source/sdk/react-native/sync-data/handle-sync-errors.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt:25]: fixed in b73e8f2a4

also made the example config use flex sync

File: source/sdk/react-native/sync-data/handle-sync-errors.txt [hunk 4]
```diff
@@ -10,20 +10,299 @@ Handle Sync Errors - React Native SDK
    :depth: 3
    :class: singlecol
 
-Overview
---------
+Handle Sync Errors
+------------------
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt]: i agree w the issue that you've diagnosed here that 'Handle Sync Errors' isn't a great title for this section. it's too similar to the page title. 

however, i don't think merging the content w the client reset intro is the right solution. client resets are really a distinct thing, and ought to be discussed as such.

i've renamed the sections 'Add a Generic Sync Error Handler' and 'Handle Client Reset Errors'. what you think of this?

File: source/sdk/react-native/sync-data/handle-sync-errors.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt:25]: Good idea on the change to flex sync! Looks like the example is still using `var`.

**Review by krollins-mdb — APPROVED** [review]
The `handle-sync-errors` example is still using `var`. Other than that, LGTM!

File: source/sdk/react-native/sync-data/handle-sync-errors.txt [hunk 5]
```diff
@@ -10,20 +10,302 @@ Handle Sync Errors - React Native SDK
...
+When Client Recovery is enabled, Realm Database automatically manages the
+client reset process in most cases. The client can recover unsynced changes
+when there are no schema changes, or non-breaking schema changes.
+
+To use Client Recovery, configure your realm with one of the following client reset modes:
+
+- Recover unsynced changes mode
+- Recover or discard unsynced changes
+
+.. include:: /includes/client-recovery-rules.rst
```
Comment (takameyer) [inline source/sdk/react-native/sync-data/handle-sync-errors.txt:113]: I think we should show the configuration in the context of the `RealmProvider`, just as we do for adding a generic sync error handler.

**Review by takameyer — APPROVED** [review]
If we change the examples to use the `RealmProvider` (or plan to do so), then I approve.

File: source/sdk/react-native/sync-data/handle-sync-errors.txt (hunk 5 above)
Comment (mongodben) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt:113]: yeah i think you're right on re-reading. i think i was just trying to rush something out here w/o enough consideration of the diff w the node.js sdk. especially b/c in the @realm/react examples we're splitting the sync config from the rest of the config.

Comment (krollins-mdb) [reply source/sdk/react-native/sync-data/handle-sync-errors.txt:113]: Ah, great point, Andrew! Would definitely be helpful to put this in the context of `<RealmProvider>`.

**Pull request merged**